app/assets/javascripts/primer_view_components.js

```diff
@@ -1 +1 @@
-
+class t extends HTMLElement{constructor(){super(),this.addEventListener("keydown",(t=>{const r=t.target;if(!(r instanceof HTMLElement))return;if("tab"!==r.getAttribute("role")&&!r.closest('[role="tablist"]'))return;const a=Array.from(this.querySelectorAll('[role="tablist"] [role="tab"]')),n=a.indexOf(a.find((t=>t.matches('[aria-selected="true"]'))));if("ArrowRight"===t.code){let t=n+1;t>=a.length&&(t=0),e(this,t)}else if("ArrowLeft"===t.code){let t=n-1;t<0&&(t=a.length-1),e(this,t)}else"Home"===t.code?(e(this,0),t.preventDefault()):"End"===t.code&&(e(this,a.length-1),t.preventDefault())})),this.addEventListener("click",(t=>{const r=Array.from(this.querySelectorAll('[role="tablist"] [role="tab"]'));if(!(t.target instanceof Element))return;const a=t.target.closest('[role="tab"]');if(!a||!a.closest('[role="tablist"]'))return;e(this,r.indexOf(a))}))}connectedCallback(){for(const t of this.querySelectorAll('[role="tablist"] [role="tab"]'))t.hasAttribute("aria-selected")||t.setAttribute("aria-selected","false"),t.hasAttribute("tabindex")||("true"===t.getAttribute("aria-selected")?t.setAttribute("tabindex","0"):t.setAttribute("tabindex","-1"))}}function e(t,e){const r=t.querySelectorAll('[role="tablist"] [role="tab"]'),a=t.querySelectorAll('[role="tabpanel"]'),n=r[e],i=a[e];if(!!t.dispatchEvent(new CustomEvent("tab-container-change",{bubbles:!0,cancelable:!0,detail:{relatedTarget:i}}))){for(const t of r)t.setAttribute("aria-selected","false"),t.setAttribute("tabindex","-1");for(const t of a)t.hidden=!0,t.hasAttribute("tabindex")||t.hasAttribute("data-tab-container-no-tabstop")||t.setAttribute("tabindex","0");n.setAttribute("aria-selected","true"),n.setAttribute("tabindex","0"),n.focus(),i.hidden=!1,t.dispatchEvent(new CustomEvent("tab-container-changed",{bubbles:!0,detail:{relatedTarget:i}}))}}window.customElements.get("tab-container")||(window.TabContainerElement=t,window.customElements.define("tab-container",t));
```

Comment:
I'm trying to think through any problems we might run into when we introduce this change to dotcom. Loading this script will increase the amount of JS loaded and will possibly load this same tab behavior twice on a page. Is that right? Anything else we could / should think about?

Reply:
This JS is not loaded every time the component is called, it is loaded either when the user adds `<%= javascript_include_tag "primer_view_components", type: "module" %>` to their view or (after we publish to NPM) adds `import @primer/view_components`

Reply:
I'm at the edge of my understanding but if a project (dotcom?) already includes `'@github/tab-container-element'`, we're still okay?

Reply:
oh, in that case it **will** be required twice
when upgrading the gem in dotcom we have to update imports to those packages

Reply:
dotcom won't be consuming this file directly, instead it will go the _other_ available route, and consume the npm package.

This file, and the `javascript_include_tag` pattern is available for those who _quickly_ want to get PVC up and running without bringing their own JS toolchain.

For those with their own JS toolchain (like dotcom), we will _also_ publish the `@primer/view_components` package. The `@primer/view_components` package will load each dependency as a module so toolchains can apply tree-shaking (aka dead/duplicate code elimination). This package can be imported with `import "@primer/view_components"` where-ever the project sees fit.

In the specific case of dotcom, we'll put it the import statement into our `behaviours` file, and our toolchain will tree-shake out any duplicate imports. It will effectively be a no-op as our `vendor` bundle already contains `@github/tab-container-element`.